ccm_web/client/src/pages/GradebookCanvas.tsx

```diff
@@ -0,0 +1,395 @@
+import React, { useEffect, useState } from 'react'
+import { Box, Button, Grid, Link, makeStyles, Paper, Typography } from '@material-ui/core'
+import CloudDoneIcon from '@material-ui/icons/CloudDone'
+import ErrorIcon from '@material-ui/icons/Error'
+import { parse, ParseResult } from 'papaparse'
+// import { useSnackbar } from 'notistack'
+
+import FileUpload from '../components/FileUpload'
+import ValidationErrorTable from '../components/ValidationErrorTable'
+import ConfirmationTable, { StudentGrade } from '../components/ConfirmationTable'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    padding: 25,
+    textAlign: 'left',
+    '& button': {
+      margin: 5
+    }
+  },
+  fileNameContainer: {
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  fileName: {
+    color: '#3F648E',
+    fontFamily: 'monospace'
+  }
+}))
+
+const useConfirmationStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  table: {
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  dialogIcon: {
+    color: '#3F648E'
+  }
+}))
+
+const useRowLevelErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  table: {
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+const useTopLevelErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10,
+    // marginLeft: 'auto',
+    // marginRight: 'auto',
+    width: '75%',
+    '& ol': {
+      margin: 'auto',
+      width: '75%'
+    },
+    '& li': {
+      textAlign: 'left'
+    }
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+interface GradebookRecord {
+  'Current Grade': string
+  'Final Grade': string
+  'SIS Login ID': string
```

Comment:
the `Current Grade here is supposed to be variables of string type. But the Current Grade is not a variable but more looks like string. Can you explain this? Sorry this might be because of my lack of enough my understand of Typescript,

Reply:
These are all the literal column names in the csv files.
The parsing library parses records into objects with these strings as property names, so you have to use this notation to reference them.

This is at least my understanding of it.

hence you muse use record['Current Grade'] instead of something more common like record.currentGrade (if the properties did not have spaces)

Reply:
OK, I will resolve this